src/autiter.rs

```diff
@@ -53,13 +53,13 @@ pub trait Automaton<P>: Sized {
     }
 
     /// Returns an iterator of non-overlapping matches in `s`.
-    fn find<'a, 's>(
+    fn find<'a, 's, Q: ?Sized + AsRef<[u8]>>(
         &'a self,
-        s: &'s str,
+        s: &'s Q,
     ) -> Matches<'a, 's, P, Self> {
```

Comment:
I couldn't get `Self: Sized` to work, but maybe I tried adding in the wrong places. But also, doesn't making things `Sized` in traits make them *less* likely to be object-safe? E.g., I just noticed that `Automaton: Sized` already, and it seems to make it already object unsafe: if I modify examples/dict-search.rs (before applying this pull request) to use `Box<Automaton>` then it fails to compile with `error: the trait `aho_corasick::autiter::Automaton` cannot be made into an object`.

Anyway, `Sized` aside, this method fails to be object-safe because object-safe traits can't have any type parameters on methods.